feat: add sample test for mobile (#35)
salesforce/utam-js-recipes: README.md

[utam-js]: https://www.github.com/salesforce/utam-js
[utam-doc]: https://utam.dev
[utam-java-recipes]: https://www.github.com/salesforce/utam-java-recipes

## Run Salesforce Mobile test

- Follow the instructions at [Get Started for Mobile](https://utam.dev/guide/get_started_utam#get-started-for-mobile) to set up your local simulator/emulator.

> - Make sure [Appium](https://github.com/appium/appium#readme) and [node.js](https://nodejs.org/en/) are installed on your machine.

Reviewer: ```suggestion
- Make sure that [Appium](https://github.com/appium/appium#readme) and [node.js](https://nodejs.org/en/) are installed on your machine.
```

- Update the wdio configuration file:

> For an iOS test, update `wdio.conf.ios.js` file to configure, test device name and os version, the full path for the test application:

Reviewer: ```suggestion
For an iOS test, update `wdio.conf.ios.js` file to configure the test device name, iOS version, and the full path for the test application:
```

Reviewer: I'm not clear which of the properties listed above correspond to the example below so it would be good to list them in the same order in the text intro and the js block below. something like:
For an iOS test, update `wdio.conf.ios.js` file to configure the test device name (`appium:deviceName`), the full path for the test application (`appium:app`), and the iOS version (`appium:platformVersion` TODO: not sure if I'm listing the correct prop here so please confirm):

> For an iOS test, update `wdio.conf.ios.js` file to configure the test device name, iOS version, and the full path for the test application:

Reviewer: Not sure if I'm matching the correct prop name in the intro to the example but will try below. Please review and then update the intro.
For an iOS test, update `wdio.conf.ios.js` file to configure the test device name (`appium:deviceName`), iOS version (`appium:platformVersion`), and the full path for the test application (`appium:app`):

```js
'appium:deviceName': 'iPhone 12',
'appium:app': '<path to iOS test app>',
'appium:platformVersion': '15.2',
```

> For an Android test, update `wdio.conf.android.js` file to configure the application bundleid, the full path for the test application and application initial activity:

Reviewer: ```suggestion
For an Android test, update the `wdio.conf.android.js` file to configure the application bundleid, the full path for the test application, and the application's initial activity:
```

Reviewer: Is application bundleid = appium:deviceName or appium:appPackage? It would be helpful to list the corresponding prop in parenthesis in the intro text like in the previous example.

We should also mention the appium:deviceName in the intro text.

I'm not sure what the "application's initial activity" means. How is it used in the test?

> For an Android test, update the `wdio.conf.android.js` file to configure the application bundleid, the full path for the test application, and the application's initial activity:

Reviewer: Not sure if I'm matching the correct prop name in the intro to the example but will try below. Please review and then update the intro.
For an Android test, update the `wdio.conf.android.js` file to configure the test device name (`appium:deviceName`), application bundleid (`appium:appPackage` TODO: not sure if I'm matching the correct prop here so please confirm), the full path for the test application (`appium:app`), and the application's initial activity (`appium:appActivity`):

Also, I'm not clear on usage of the application's initial activity so please explain its usage in the test

```js
'appium:deviceName': 'emulator-5554',
'appium:app': '<path to Android test app>',
'appium:appActivity': 'com.salesforce.chatter.Chatter',
'appium:appPackage': 'com.salesforce.chatter',
```

> - Download the [debug build](https://developer.salesforce.com/tools/mobile-debugging) for SalesforceApp iOS and Android.

Reviewer: ```suggestion
- Download the [debug build](https://developer.salesforce.com/tools/mobile-debugging) for the Salesforce iOS and Android mobile apps.
```

> - Commands to execute test: 

Reviewer: ```suggestion
- Commands to execute a test: 
```

For iOS: yarn test wdio.conf.ios.js
For Android: yarn test wdio.conf.android.js

> - For a test on Android, make sure to start an emulator before the test run. Otherwise, will hit error like this: "Error: Failed to create session.

Reviewer: ```suggestion
- For a test on Android, make sure to start an emulator before the test run. Otherwise, you will get an error like this: "Error: Failed to create session.
```

[utam-js]: https://www.github.com/salesforce/utam-js
[utam-doc]: https://utam.dev
[utam-java-recipes]: https://www.github.com/salesforce/utam-java-recipes

## Run Salesforce Mobile test

- Follow the instructions at [Get Started for Mobile](https://utam.dev/guide/get_started_utam#get-started-for-mobile) to set up your local simulator/emulator.

> - Make sure Appium and Nodejs are installed in user home.

Reviewer 2: nit: typo it's node.js

Reviewer 2: suggestion: I would add the link for both project homepage

> - Update the wdio configure file:

Reviewer 2: nit: typo `configure` -> `configuration`

> - Make sure [Appium](https://github.com/appium/appium#readme) and [node.js](https://nodejs.org/en/) are installed in user home.

Reviewer 2: nit: does it matter that they are installed in the user's home folder? I can tell it doesn't for node but I have no idea for appium

> For an iOS test, update wdio.conf.ios.js file to configure, test device name and os version, the full path for the test application:

Reviewer 2: suggestion: enclose the config filename in backticks

```js
'appium:deviceName': 'iPhone 12',
'appium:app': '<path to iOS test app>',
'appium:platformVersion': '15.2',
```

> For an Android test, update wdio.conf.android.js file to configure the application bundleid, the full path for the test application and application initial activity:

Reviewer 2: suggestion: enclose the config filename in backticks

```js
'appium:deviceName': 'emulator-5554',
'appium:app': '<path to Android test app>',
'appium:appActivity': 'com.salesforce.chatter.Chatter',
'appium:appPackage': 'com.salesforce.chatter',
```

- Download the [debug build](https://developer.salesforce.com/tools/mobile-debugging) for SalesforceApp iOS and Android.
- Commands to execute test: 
For iOS: yarn test wdio.conf.ios.js
For Android: yarn test wdio.conf.android.js

> - Start an Appium server at local before executing any test via: appium --port 4444.

Reviewer 2: nit: I would hoist that before the test commands to make sure people read the info in the order they have to execute them
